### PR TITLE
ignore system test gems on Gemfile when execute with --skip-test option

### PR DESCRIPTION
**Pull request by ota42y**
### Summary
`rails new app_name -T` generate rails application without test.
But, generated Gemfile include `capybara` and `selenium-webdriver`.
So I fix don't generate these gems when test skip option add.

### CODE REVIEW

File: railties/lib/rails/generators/rails/app/templates/Gemfile [hunk 1]
```diff
@@ -32,7 +32,7 @@ end
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
-  <%- unless options.skip_system_test? || options.api? -%>
+  <%- unless options.skip_system_test? || options.skip_test? || options.api? -%>
```
Comment (rafaelfranca) [inline railties/lib/rails/generators/rails/app/templates/Gemfile]: Can we create a method like `depend_on_listen` and move those conditionals there?

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: railties/lib/rails/generators/rails/app/templates/Gemfile (hunk 1 above)
Comment (ota42y) [inline railties/lib/rails/generators/rails/app/templates/Gemfile]: I did fix!

File: railties/lib/rails/generators/app_base.rb [hunk 2]
```diff
@@ -413,6 +413,10 @@ def spring_install?
         !options[:skip_spring] && !options.dev? && Process.respond_to?(:fork) && !RUBY_PLATFORM.include?("cygwin")
       end
 
+      def depend_system_test?
```
Comment (eileencodes) [inline railties/lib/rails/generators/app_base.rb]: Can you make this `depend_on_system_test?` so it matches other methods in this file like `depend_on_listen?`

Comment (ota42y) [inline railties/lib/rails/generators/app_base.rb]: fixed!

**Comment by eileencodes** [comment]
Looks good. Can you squash your commits into one? Thanks!

**Comment by ota42y** [comment]
I did squash!

**Comment by eileencodes** [comment]
Thanks @ota42y! 😄

**Comment by eileencodes** [comment]
Backported to 5-1-stable in 716a9fbae7ed89df807143a84eb104ba82f8bc04

**Pull request merged**